File: src/main/javabak/tile_entities/capabilities/SimpleDroppingContainer.java
package redcrafter07.processed.block.tile_entities.capabilities;

import net.minecraft.world.SimpleContainer;
import net.minecraft.world.item.ItemStack;

public class SimpleDroppingContainer extends SimpleContainer {
    public SimpleDroppingContainer(ItemStack... items) {
        super(items);
    }

    public void appendItem(ItemStack itemStack) {
        var stack = super.addItem(itemStack);
        if (!stack.isEmpty()) getItems().add(stack);
    }
}

File: src/main/kotlin/redcrafter07/processed/block/tile_entities/capabilities/SimpleDroppingContainer.kt
package redcrafter07.processed.block.tile_entities.capabilities

import net.minecraft.world.SimpleContainer
import net.minecraft.world.item.ItemStack

class SimpleDroppingContainer() : SimpleContainer() {
    fun appendItem(itemStack: ItemStack) {
        val stack = super.addItem(itemStack)
        if (!stack.isEmpty) items.add(stack)
    }
}